Vladlan/nodejs-aws-be
# Task 9

- **FE link**: https://d32m2krhvblmxn.cloudfront.net/
- **product-service** service API endpoint URL - https://269pjntj1e.execute-api.eu-west-1.amazonaws.com/dev/products
- example of the **create product** API call with all needed information: URL, payload, headers, etc:
`
curl --location --request POST 'https://269pjntj1e.execute-api.eu-west-1.amazonaws.com/dev/products' \
--header 'Content-Type: application/json' \
--data-raw '{
    "count": 1,
    "description": "Test product 1 description",
    "price": 1,
    "title": "Test product 1"
}'
`
- **CART** service API endpoint URL - http://vladlan-cart-api-develop.eu-west-1.elasticbeanstalk.com/api/profile/cart
- **bff-service** service URL - http://vladlan-bff-api-develop-bff-service.eu-west-1.elasticbeanstalk.com
- example how to call **product-service** and **CART** services via **bff-service** service URL:
get products - http://vladlan-bff-api-develop-bff-service.eu-west-1.elasticbeanstalk.com/product/products/
get cart - http://vladlan-bff-api-develop-bff-service.eu-west-1.elasticbeanstalk.com/cart/api/profile/cart
---

**What has been done:**

**3** - A working and correct **express** application should be in the **bff-service** folder. Reviewer can start this application locally with any valid configuration in the **.env** file and this application should works as described in the task 9.1

**5** - The **bff-service** should be deployed with Elastic Beanstalk. The **bff-service** call should be redirected to the appropriate service : **product-service** or **CART**. The response from the **bff-service** should be the same as if **product-service** or **CART** services were called directly.
 
## Additional (optional) tasks
---
**+1** - Add a cache at the **bff-service** level for a request to the **getProductsList** function of the **product-service**. The cache should expire in 2 minutes.  
How to test:
  * Get products list
  * Create new product
  * Get products list - result shouldn’t have new product
  * Wait more than 2 minutes
  * Get products list - result should have new product

**+1** - Use **NestJS** to create **bff-service** instead of **express**

---
**Final mark: 7/7**
